orangecontrib/text/widgets/owcorpus.py

```diff
@@ -303,6 +312,60 @@ def describe(features):
                 ('Target', describe(domain.class_vars)),
             ))
 
+    def handle_languages(self):
+        if self.corpus is not None:
+            domain = self.corpus.domain
+            if self.detect_languages:
+                if self.corpus.languages is None:
+                    self.corpus.detect_languages()
+
+                curr_attributes = list(domain.attributes)
+                curr_class_var = [domain.class_var] if domain.class_var else []
+                curr_metas = list(domain.metas)
+                curr_variables = curr_attributes + curr_class_var + curr_metas
+                curr_names = [var.name for var in curr_variables]
```

Comment:
I believe you can make this shorter with:
`[var.name for var in domain.variables]`
without any of the calls above.

Reply:
This doesn't seem to include names of metas.

Reply:
You're right. I hate this. But keeping with a one-lines, this should work:
`[var.name for var in corpus.domain.variables + corpus.domain.metas]`

No need to fix it now, I just think it is easier to read and keeps the code base concise. :)